mlflow/gateway/providers/bedrock.py

```diff
@@ -0,0 +1,279 @@
+            return False
+
+        return (
+            (time.monotonic_ns() - self._client_created)
+            >= (self.bedrock_config.aws_config.session_length_seconds) * 1_000_000_000,
+        )
+
+    def get_bedrock_client(self):
+        if self._client is not None and not self._client_expired():
+            return self._client
+
+        session_args, config_args, client_args = {}, {}, {}
+        aws_config = self.bedrock_config.aws_config
+
+        region_name = aws_config.aws_region or os.environ.get(
+            "AWS_REGION", os.environ.get("AWS_DEFAULT_REGION")
+        )
+
+        if region_name:
+            config_args["region_name"] = session_args["region_name"] = region_name
+
+        # check profile, mostly for dev
+        profile_name = os.environ.get("AWS_PROFILE")
+        if profile_name:
+            session_args["profile_name"] = profile_name
+
+        session = boto3.Session(**{k: v for k, v in session_args.items() if v is not None})
+
+        if isinstance(aws_config, AWSRole):
+            role = session.client(service_name="sts").assume_role(
+                RoleArn=aws_config.aws_role_arn,
+                RoleSessionName="ai-gateway-bedrock",
+                DurationSeconds=aws_config.session_length_seconds,
+            )
+            client_args["aws_access_key_id"] = role["Credentials"]["AccessKeyId"]
+            client_args["aws_secret_access_key"] = role["Credentials"]["SecretAccessKey"]
+            client_args["aws_session_token"] = role["Credentials"]["SessionToken"]
+        elif isinstance(aws_config, AWSIdAndKey):
+            client_args["aws_access_key_id"] = aws_config.aws_access_key_id
+            client_args["aws_secret_access_key"] = aws_config.aws_secret_access_key
+            client_args["aws_session_token"] = aws_config.aws_session_token
+        else:
+            # defer to standard CredentialProviderChain
+            pass
+
+        if config_args:
+            client_args["config"] = botocore.config.Config(**config_args)
+
+        try:
+            # use default/prevailing credentials
+            self._client, self._client_created = (
+                session.client(service_name="bedrock", **client_args),
+                time.monotonic_ns(),
+            )
+            return self._client
+
+        except botocore.exceptions.UnknownServiceError as e:
+            raise AIGatewayConfigException(
+                "Cannot create AWS Bedrock client; ensure boto3/botocore "
+                "linked from the AWS Bedrock user guide are installed. "
+                "Otherwise likely missing credentials or accessing account without to "
+                "AWS Bedrock Private Preview"
+            ) from e
+
+    @property
+    def _underlying_provider(self):
+        if (not self.config.model.name) or "." not in self.config.model.name:
+            return None
+        provider, *_ = self.config.model.name.split(".")
```

Comment:
If we're just interested in the first element of the split, can we simplify this without providing an anonymous list capture here?

```python
@property
def _underlying_provider(self):
    model_name = self.config.model.name
    if not model_name or "." not in model_name:
        return None

    provider = model_name.split(".")[0]
    return AWSBedrockModelProvider.of_str(provider)
```